k2/csrc/weights.cc

```diff
@@ -0,0 +1,81 @@
+// k2/csrc/weights.cc
+
+// Copyright (c)  2020  Xiaomi Corporation (author: Haowen Qiu)
+
+// See ../../LICENSE for clarification regarding multiple authors
+
+#include "k2/csrc/weights.h"
+
+#include <algorithm>
+#include <queue>
+#include <vector>
+
+#include "glog/logging.h"
+#include "k2/csrc/fsa.h"
+#include "k2/csrc/properties.h"
+
+namespace {
+void CheckInput(const k2::Fsa &fsa, const float *arc_weights) {
+  CHECK(IsValid(fsa));
+  CHECK_NOTNULL(arc_weights);
+}
+}  // namespace
+
+namespace k2 {
+
+void ComputeForwardMaxWeights(const Fsa &fsa, const float *arc_weights,
+                              float *state_weights) {
+  if (IsEmpty(fsa)) return;
+  CheckInput(fsa, arc_weights);
+  CHECK_NOTNULL(state_weights);
+
+  int32_t num_states = fsa.NumStates();
+  std::fill_n(state_weights, num_states, kNegativeInfinity);
+
+  const auto &arcs = fsa.arcs;
+  state_weights[0] = 0;
+  for (std::size_t i = 0; i != arcs.size(); ++i) {
+    const auto &arc = arcs[i];
+#ifndef NDEBUG
```

Comment:
please use `DCHECK_GE`, no need to guard it with `NDEBUG`.

Refer to https://github.com/google/glog/blob/3ba8976592274bc1f907c402ce22558011d6fc5e/src/glog/logging.h.in#L438

```cpp
#if defined(NDEBUG) && !defined(DCHECK_ALWAYS_ON)
#define DCHECK_IS_ON() 0
#else
#define DCHECK_IS_ON() 1
#endif
```

and

https://github.com/google/glog/blob/master/src/glog/logging.h.in#L1004

Reply:
Thanks